sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/LDeviceAdapter.java

```diff
@@ -359,6 +377,74 @@ protected boolean hasControlBlockCreationCapability(ControlBlockEnum controlBloc
 
     private boolean hasCBNameConf(TServiceSettings tServiceSettings) {
         return tServiceSettings.isSetCbName()
-            && (TServiceSettingsNoDynEnum.CONF.equals(tServiceSettings.getCbName()));
+                && (TServiceSettingsNoDynEnum.CONF.equals(tServiceSettings.getCbName()));
+    }
+
+    /**
+     * Update and/or create Monitoring LNs (LSVS and LGOS) into LDSUIED from ExtRefs binding
+     *
+     * @param tExtRefs           ExtRefs for which source Control Blocks (Goose or SMV) should be monitored
+     * @param tSystemLNGroupEnum LNClass value for LN to monitor (Only LGOS and LSVS are taken into this enum)
+     * @return a list of SclReport Objects that contains errors
+     */
+    public List<SclReportItem> manageMonitoringLns(List<TExtRef> tExtRefs, String doName, TSystemLNGroupEnum tSystemLNGroupEnum) {
+        if (tExtRefs.isEmpty() || (!TSystemLNGroupEnum.LGOS.equals(tSystemLNGroupEnum)
+                && !TSystemLNGroupEnum.LSVS.equals(tSystemLNGroupEnum))) {
```

Comment:
I don't understand why we filter `tSystemLNGroupEnum` parameter here ? The parameter is hardcoded each time the method is called (first time it is `TSystemLNGroupEnum.LGOS` and second time `TSystemLNGroupEnum.LSVS`). The test on `tSystemLNGroupEnum` is always false.
I suggest removing the test (keep only the test on tExtRefs)

Reply:
ok